aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -27,37 +27,196 @@ Kestrel supports the following features:
+
+**Maximum client connections**
+
+The maximum number of concurrent open TCP connections can be set for the entire application with the following code:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=3-4)]
+
+There's a separate limit for connections that have been upgraded from HTTP or HTTPS to another protocol (for example, on a WebSockets request). After a connection is upgraded, it isn't counted against the `MaxConcurrentConnections` limit. 
+
+The maximum number of connections is unlimited (null) by default.
+
+**Maximum request body size**
+
+The default maximum request body size is 30,000,000 bytes, which is approximately 28.6MB. 
+
+The recommended way to override the limit in an ASP.NET Core MVC app is to use the [RequestSizeLimit](https://github.com/aspnet/Mvc/blob/rel/2.0.0/src/Microsoft.AspNetCore.Mvc.Core/RequestSizeLimitAttribute.cs) attribute on an action method:
+
+```csharp
+[RequestSizeLimit(100000000)]
+public IActionResult MyActionMethod()
+```
+
+Here's an example that shows how to configure the constraint for the entire application, every request:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=5)]
+
+You can also override the setting on a specific request in the `Configure` method of *Startup.cs*:
+
+[!code-csharp[](kestrel/sample2/Startup.cs?name=snippet_Limits&highlight=3-4)]
+ 
+An exception is thrown if you try to configure the limit on a request after the application has started reading the request. There's an `IsReadOnly` property that tells you if the `MaxRequestBodySize` property is in read-only state, meaning it's too late to configure the limit.
+
+**Minimum request body data rate**
+
+Kestrel checks every second if data is coming in at the specified rate in bytes/second. If the rate drops below the minimum, the connection is timed out. The grace period is the amount of time that Kestrel gives the client to increase its send rate up to the minimum; the rate is not checked during that time. The grace period helps avoid dropping connections that are initially sending data at a slow rate due to TCP slow-start.
+
+The default minimum rate is 240 bytes/second, with a 5 second grace period.
+
+Here's an example that shows how to configure the minimum request rate in *Program.cs*:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=6-7)]
+
+You can configure the rate per request in *Startup.cs*:
+
+[!code-csharp[](kestrel/sample2/Startup.cs?name=snippet_Limits&highlight=5-6)]
+
+You can also set the minimum data rate for the response. The property and interface names have `Response` in place of `RequestBody`; otherwise the code to configure this limit is the same. 
```

Comment:
Can we actually show this in one of the samples? I feel that both would often be configured together.